### PR TITLE
Classifier: add survey task choice close/delete on Escape key

### LINKED ISSUE #3831: Survey task should close Choice, show Chooser, on 'Escape' keydown.
From a quick read of the code yesterday, the `Choice` component could call `handleDelete(choiceId)` on ‘Esc’. That doesn’t seem hard. That could be handled either in the `Choice` component or in `SurveyTask`.

There’s a reference to `onEsc` in `handleChoice` but not in `handleDelete`. 
https://github.com/zooniverse/front-end-monorepo/blob/3b63c6c2393475d848d24ad96b14c35bf6d3ecf0/packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTaskContainer.js#L51

_Originally posted by @eatyourgreens in https://github.com/zooniverse/front-end-monorepo/issues/3735#issuecomment-1294544018_

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
- closes #3831

## Describe your changes
- adds onKeyDown to Choice component, if Escape key then delete/close choice
- adds focus to Choice component on open
- adds `aria-labelledby` for Choice component
- adds `aria-labelledby` for choice question input groups

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=1634&workflow=2434
- What user actions should my reviewer step through to review this PR?
  - open a survey task choice, tab/click in choice, press Escape key
- Which storybook stories should be reviewed?
  - http://localhost:6006/?path=/story/tasks-survey--no-filters
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js [hunk 1]
```diff
@@ -180,6 +180,17 @@ describe('SurveyTask with user keystrokes', function () {
       expect(choiceImages).to.be.ok()
     })
 
+    it('should close choice on Escape key', async function () {
+      // tabbing to the first choice (Aardvark) and pressing Enter to open the choice (Aardvark)
+      await user.keyboard('[Enter]')
+      // tabbing into the choice component
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js:186]: Shouldn’t the choice receive focus as soon as it opens? How would a blind volunteer use this, on an audio survey workflow?

**Review by eatyourgreens — APPROVED** [review]
Looks good but I’m not sure why you have to press Tab before you can use Esc. That’s not intuitive, particularly if you can’t see the page.

File: packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js]: Added focus to the choice component, `aria-labelledby` for the focused component, and `aria-labelledby` for the choice question input groups. I tested with VoiceOver and navigating into and within the Choice seems improved to me, but please let me know what you think (no rush).

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js [hunk 2]
```diff
@@ -23,22 +31,39 @@ export default function Choice({
...
+    if (event.key === 'Escape') {
+      handleDelete(choiceId)
+    }
+  }
+
   return (
-    <Box
+    <StyledBox
+      ref={choiceRef}
+      aria-labelledby='choice-label'
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:57]:
```suggestion
      aria-labelledby='choice-label'
      as=‘section’
```
I’d recommend using `section` here as `div` can’t be labelled.

**Comment by eatyourgreens** [comment]
FWIW, I think you found a bug in VoiceOver while testing this. The ARIA spec forbids [labels on generic elements](https://html5accessibility.com/stuff/2022/04/03/aria-labelledby-usage-notes/).

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js [hunk 3]
```diff
@@ -23,22 +31,40 @@ export default function Choice({
...
+      as=‘section’
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Choice/Choice.js:58]:
```suggestion
      as='section'
```

**Review by eatyourgreens — APPROVED** [review]

File: packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.userKeystrokes.spec.js]: Yes it should, great catch! I updated code to focus Choice component on open.

**Pull request merged**